extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,175 @@
+package org.schabi.newpipe.extractor.services.youtube.extractors;
+
+import static org.schabi.newpipe.extractor.services.youtube.linkHandler.YoutubeParsingHelper.getJsonResponse;
+
+import com.grack.nanojson.JsonArray;
+import com.grack.nanojson.JsonObject;
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.schabi.newpipe.extractor.ListExtractor;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+import org.schabi.newpipe.extractor.localization.TimeAgoParser;
+import org.schabi.newpipe.extractor.playlist.PlaylistExtractor;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector;
+
+/**
+ * A YoutubePlaylistExtractor for a mix (auto-generated playlist). It handles urls in the format of
+ * "youtube.com/watch?v=videoId&list=playlistId"
+ */
+public class YoutubeMixPlaylistExtractor extends PlaylistExtractor {
+
+
+    private final static String CONTENTS = "contents";
+    private final static String RESPONSE = "response";
+    private final static String PLAYLIST = "playlist";
+    private final static String TWO_COLUMN_WATCH_NEXT_RESULTS = "twoColumnWatchNextResults";
+    private final static String PLAYLIST_PANEL_VIDEO_RENDERER = "playlistPanelVideoRenderer";
```

Comment:
Why are you storing those as constants here? We don't do that anywhere else either…

Reply:
Because i would sometime make a typo in those strings and wonder for 10 minutes where my error was. With the constants i can let my IDE autocomplete for me.

Reply:
Please just follow the style of the rest of the extractors.